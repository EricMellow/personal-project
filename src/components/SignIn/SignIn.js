import React, { Component } from "react";
import './SignIn.css';
import { auth } from '../../firebase';

export class SignIn extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      error: null
    };
  }

  handleInput = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  resetState = () => {
    this.setState({
      email: '',
      password: '',
      error: null
    });
  }

  storeData = async (event) => {
    event.preventDefault();
    try {
      await auth.doSignInWithEmailAndPassword(this.state.email, this.state.password);
      this.resetState();
      this.props.history.push('/distance');
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  render() {
    const {
      email,
      password,
      error
    } = this.state;

    const invalidUserInfo =
      password === '' ||
      email === '';

    return (
      <form onSubmit={event => this.storeData(event)}>
        <input
          placeholder="Email Address"
          name="email"
          className="input email-input"
          value={email}
          onChange={event => this.handleInput(event)}
          type="text"
        />
        <input
          placeholder="Password"
          name="password"
          className="input password-input"
          value={password}
          onChange={event => this.handleInput(event)}
          type="password"
        />
        <button type="submit" disabled={invalidUserInfo}>
          Sign In
        </button>

        {error && <p>{error}</p>}
      </form>
    );
  }
}

export default SignIn;