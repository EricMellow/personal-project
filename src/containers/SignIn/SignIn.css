.sign-in {
  align-items: center;
  background-repeat: no-repeat;
  background-size: cover;
  background: url(../../assets/bricks1.jpg); 
  display: flex;
  flex-direction: column;
  height: 85vh;
  justify-content: center;
}

.sign-in-form {
  align-items: center;
  background-color: rgba(28, 141, 196, .9);
  display: flex;
  flex-direction: column;
  height: 55%;
  justify-content: center;
  width: 50%;
}

.input {
  background: none;
  border-radius: 5px;
  border: 3px solid white;
  color: white;
  font-size: 2rem;
  height: 3rem;
  margin: 20px;
  padding-left: 10px;
  width: 75%;
}

.sign-in-btn {
  background: none;
  border-radius: 5px;
  border: 2px solid white;
  color: white;
  font-size: 2rem;
  height: 3rem;
  margin: 20px;
  width: 75%;
}

.sign-in-btn:disabled {
  background: rgba(255, 255, 255, 0.15);
  border: 2px solid rgb(67, 119, 144);
  color: rgb(67, 119, 144);
}

*, *:before, *:after {
  box-sizing: inherit;
}

form p {
  color: white;
  font-size: 1.2rem;
  margin: 0;
}