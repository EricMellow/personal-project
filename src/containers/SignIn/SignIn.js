import React, { Component } from "react";
import './SignIn.css';
import { auth } from '../../firebase';
import { connect } from 'react-redux';
import { authenticateUser } from "../../actions/authenticateUser";
import { addUserId } from "../../actions/userIdActions";
import { addZipcode } from "../../actions/zipcodeActions";
import * as firebase from "../../firebase/firebase";

export class SignIn extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      error: null
    };
  }

  handleInput = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  resetState = () => {
    this.setState({
      email: '',
      password: '',
      error: null
    });
  }

  storeData = async (event) => {
    event.preventDefault();
    try {
      const authUser = await auth.doSignInWithEmailAndPassword(this.state.email, this.state.password);
      this.props.storeUserId(authUser.user.uid);
      this.resetState();
      this.props.authenticate();
      firebase.db.ref(`users/${authUser.user.uid}`).once('value')
        .then(snapshot => this.props.storeZipcode(snapshot.val().zipcode));
      this.props.history.push('/distance');
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  render() {
    const {
      email,
      password,
      error
    } = this.state;

    const invalidUserInfo =
      password === '' ||
      email === '';

    return (
      <div className="sign-in">
        <form className="sign-in-form" onSubmit={event => this.storeData(event)}>
          <input
            placeholder="Email Address"
            name="email"
            className="input email-input"
            value={email}
            onChange={event => this.handleInput(event)}
            type="text"
          />
          <input
            placeholder="Password"
            name="password"
            className="input password-input"
            value={password}
            onChange={event => this.handleInput(event)}
            type="password"
          />
          <button 
            type="submit" 
            className="sign-in-btn"
            disabled={invalidUserInfo}
          >
            Sign In
          </button>

          {error && <p>{error}</p>}
        </form>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  authenticate: () => dispatch(authenticateUser()),
  storeUserId: (userId) => dispatch(addUserId(userId)),
  storeZipcode: (zipcode) => dispatch(addZipcode(zipcode))
});

export default connect(null, mapDispatchToProps)(SignIn);