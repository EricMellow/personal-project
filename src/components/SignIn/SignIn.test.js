import SignIn from "./SignIn";
import { shallow } from "enzyme";
import React from "react";

describe('SignIn', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SignIn />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleInput', () => {

    it('should set the correct key/value pair in state with the incomming event data', () => {
      const wrapper = shallow(<SignIn />);
      const mockEvent = {
        target: {
          name: 'password',
          value: 'trustnoone'
        }
      };
      const initialExpected = {
        email: '',
        password: '',
        error: null
      };
      const finalExpected = {
        email: '',
        password: 'trustnoone',
        error: null
      };

      expect(wrapper.state()).toEqual(initialExpected);
      wrapper.instance().handleInput(mockEvent);
      expect(wrapper.state()).toEqual(finalExpected);
    });
  });

  describe('resetState', () => {

    it('should reset the state to the default settings', () => {
      const wrapper = shallow(<SignIn />);
      const initialState = {
        email: 'test@test.com',
        password: 'password',
        error: null
      };
      wrapper.setState(initialState);
      const expected = {
        email: '',
        password: '',
        error: null
      };

      expect(wrapper.state()).toEqual(initialState);
      wrapper.instance().resetState();
      expect(wrapper.state()).toEqual(expected);
    });
  });

});