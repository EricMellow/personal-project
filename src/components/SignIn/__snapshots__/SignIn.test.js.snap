// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          type="text"
          value=""
/>,
        <input
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
/>,
        <button
          disabled={true}
          type="submit"
>
          Sign In
</button>,
        null,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "email",
          "onChange": [Function],
          "placeholder": "Email Address",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "password",
          "onChange": [Function],
          "placeholder": "Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign In",
          "disabled": true,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Sign In",
        "type": "button",
      },
      null,
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            type="text"
            value=""
/>,
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
/>,
          <button
            disabled={true}
            type="submit"
>
            Sign In
</button>,
          null,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email Address",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign In",
            "disabled": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": "button",
        },
        null,
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
